src/net/socket.rs

```diff
@@ -159,29 +132,37 @@ impl Socket {
 
     // Receives a single message from the socket. On success, returns the packet containing origin and data.
     fn recv_from(&mut self) -> Result<Option<Packet>> {
-        let (recv_len, address) = self.socket.recv_from(&mut self.recv_buffer)?;
-        if recv_len == 0 {
-            return Err(ErrorKind::ReceivedDataToShort)?;
+        match self.socket.recv_from(&mut self.recv_buffer) {
+            Ok((recv_len, address)) => {
+                if recv_len == 0 {
+                    return Err(ErrorKind::ReceivedDataToShort)?;
+                }
+                let received_payload = &self.recv_buffer[..recv_len];
+                let connection = self
+                    .connections
+                    .get_or_insert_connection(address, &self.config);
+                connection.process_incoming(received_payload)
+            }
+            Err(ref e) if e.kind() == io::ErrorKind::WouldBlock => {
+                error!("Encountered a WouldBlock: {:?}", e);
```

Comment:
This should likely just return if the socket would have been blocked. Also, we probably don't want to lose any of the packets pulled off of the channel in the caller so it may make sense to bubble up the error and handle it at the call site.

Reply:
Sorry, not sure what you mean. If it hits the WouldBlock arm, it will return. If we want it to bubble up the error, we'll have to wrap the socket::io error types with our own error type.

Reply:
You know what, I take it back. I believe I was mistaking this for the send method. This is fine.